### PR TITLE
Fix agility laps left counter

### LINKED ISSUE #2282: Agility plugin "Laps till level" disappears on level up
Per the below screenshots, the "Laps till level" display disappears on level-up. It only re-appears after completing another lap after reaching the new level.

agility 83

### PR DESCRIPTION
**Pull request by AWarbear**
Stop the laps remaining overlay from disappearing on level up at end of course (Fixes #2282 )
Make the laps remaining counter count the laps remaining to the next level if the previous next level was reached with the exp gained.
Select the active course based on regionId as the results from selecting by exp are unreliable because there are courses that give uneven amounts of exp/lap

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java [hunk 1]
```diff
@@ -56,6 +56,13 @@ public void incrementLapCount(Client client)
 		int nextLevel = client.getRealSkillLevel(Skill.AGILITY) + 1;
 		int remainingXp = nextLevel <= Experience.MAX_VIRT_LEVEL ? Experience.getXpForLevel(nextLevel) - currentExp : 0;
 
+		//If next level is reached, count to the level after
+		if (remainingXp < 0)
+		{
+			nextLevel += 1;
+			remainingXp = nextLevel <= Experience.MAX_VIRT_LEVEL ? Experience.getXpForLevel(nextLevel) - currentExp : 0;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java]: Make a function for this? It's exactly the same code as L57.

Comment (AWarbear) [reply runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java]: Changed it to while loop, as it occurred to me that in things like DMM one might level up more than one level upon course completion

File: runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java [hunk 2]
```diff
@@ -54,7 +55,13 @@ public void incrementLapCount(Client client)
...
 		int currentExp = client.getSkillExperience(Skill.AGILITY);
 		int nextLevel = client.getRealSkillLevel(Skill.AGILITY) + 1;
-		int remainingXp = nextLevel <= Experience.MAX_VIRT_LEVEL ? Experience.getXpForLevel(nextLevel) - currentExp : 0;
+
+		int remainingXp;
+		do
+		{
+			remainingXp = nextLevel <= Experience.MAX_VIRT_LEVEL ? Experience.getXpForLevel(nextLevel) - currentExp : 0;
+			nextLevel++;
+		} while (remainingXp < 0);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java]: What is this for? The xp is being added before the level is increased?

Comment (AWarbear) [reply runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java]: I'm not sure I understand the question, but it happens that getSkillExperience returns more exp than what is the exp of getRealSkillLevel + 1 this happens if the player levels up when completing the lap.
Next level starts at getRealSkillLevel + 1 before the loop and is increased until either the level is more than currentExp that is gotten from getSkillExperience or the level is greater or equal than MAX_VIRT_LEVEL.

**Pull request merged**